src/commands/login.ts

```diff
@@ -0,0 +1,191 @@
+import { Command, Flags, ux } from '@oclif/core';
+import { AuthService } from '../services/auth.service';
+import { ValidationService } from '../services/validation.service';
+import {
+  EmptyPasswordError,
+  NoFlagProvidedError,
+  NotValidEmailError,
+  NotValidTwoFactorCodeError,
+} from '../types/login.types';
+import { ConfigService } from '../services/config.service';
+import { CLIUtils } from '../utils/cli.utils';
+
+export default class Login extends Command {
+  static readonly args = {};
+  static readonly description =
+    'Logs into an Internxt account. If the account is two-factor protected, then an extra code will be required.\n' +
+    'Using the password parameter is not recommended as it can lead to security problems (the password is written plainly in the console), ' +
+    'it is safer to type your password interactively when the cli asks for it.';
+
+  static readonly examples = ['<%= config.bin %> <%= command.id %>'];
+
+  static readonly flags = {
+    email: Flags.string({
+      char: 'e',
+      env: 'INXT_EMAIL',
+      description: 'The email to log in',
+      required: false,
+    }),
+    password: Flags.string({
+      char: 'p',
+      env: 'INXT_PASSWORD',
+      description: '[Insecure] The plain password to log in',
+      required: false,
+    }),
+    'two-factor': Flags.string({
+      char: 'w',
+      env: 'INXT_TWOFACTORCODE',
+      description: '[If needed] The two factor auth code',
+      required: false,
+      helpValue: '123456',
+    }),
+    'non-interactive': Flags.boolean({
+      char: 'n',
+      env: 'INXT_NONINTERACTIVE',
+      helpGroup: 'helper',
+      description:
+        'Blocks the cli from being interactive. If passed, the cli will not request data through the console and will throw errors directly',
+      required: false,
+    }),
+  };
+
+  public async run(): Promise<void> {
+    const { flags } = await this.parse(Login);
+
+    const email =
+      (await this.getEmailFromFlag(flags['email'], flags['non-interactive'])) ?? (await this.getEmailInteractively());
```

Comment:
But what's the point of mixing both modes? so you can for example do this?

`internxt-cli login --email=email@email.com` and then it will ask you interactively the rest of variables (password and 2FA)?